src/tools/qt-gui/src/guisettings.cpp

```diff
@@ -0,0 +1,162 @@
+#include "guisettings.hpp"
+#include <QDebug>
+#include <QPalette>
+
+using namespace kdb;
+
+namespace kdb
+{
+template <>
+inline QColor Key::get() const
+{
+	if (getStringSize() < 1)
+	{
+		throw KeyTypeConversion();
+	}
+
+	std::string str = getString();
+	QColor c(str.c_str());
+	return c;
+}
+}
+
+GUISettings::GUISettings(QObject *parentGUISettings) : QObject(parentGUISettings)
+{
+	QPalette palette;
+	palette.setCurrentColorGroup(QPalette::Active);
+
+	//initialize default colors
+	m_highlightColor		= palette.highlight().color();
+	m_frameColor			= palette.dark().color();
+	m_nodeWithKeyColor		= palette.windowText().color();
+
+	palette.setCurrentColorGroup(QPalette::Disabled);
+
+	m_nodeWithoutKeyColor	= palette.windowText().color();
+
+	//set base path for all colors
+	m_base = "/sw/libelektra.org/qt-gui/#0/";
+
+	try
+	{
+		m_kdb.get(m_config, m_base);
+	}
+	catch(const KDBException &ex)
+	{
+		qDebug() << ex.what();
+	}
+
+	//check if stored colors exist, if so, load them,else create them
+	if(!lookup("highlightColor").isValid())
+		append("highlightColor", m_highlightColor);
+	else
+		m_highlightColor = lookup("highlightColor");
```

Comment:
avoid that these strings are present multiple times. Also make them lowercase with underscores.